### PR TITLE
Release/week1

### PR DESCRIPTION
**Pull request by saifmohammed888**
Week 1 tasks

Frontend setup
Backend Setup
Configure backend server (Heroku)
Configure frontend server (Netlify)
Database design and Setup
Setup CI/CD (Frontend and backend)
Landing Page
Login and Signup Backend
Login and SignUp Frontend
Admin Dashboard
Resource Page
Interviewer Onboard Page
Topics Page
Student and Interviewer Profile
Quiz
Admin Routes
Additional Pages
Additional Pages

### CODE REVIEW

File: src/components/Dashboard/Admin/Dashboard.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+                </section>
+                <section className="resourceChart">
+                    <DashboardPieChart options={resourceChartOptions} data={resourceChartData} />
+                </section>
+            </section>
+
+            <section className="customer-feedback">
+                <section className="total-earning">
+                    <section className="top">
+                        <p>Total Earning</p>
```
Comment (sraghudatta94) [inline src/components/Dashboard/Admin/Dashboard.js:35]: from const, so thatit will be easy for api integration

File: src/components/Dashboard/Admin/Dashboard.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+                        <p>Total Sales</p>
+                        <h4>₹87,493</h4>
+                        <p>5.43% Since Last Month</p>
+                    </section>
+                    <DashboardChart className="totalEarningChart" options={totalSaleOption} data={totalSaleData} />
+                </section>
+                <section className="cust-review">
+                    {CustomerReviews.map((review,index) => (
+                        <section key={index} className="review-container">
+                            <span style={{display:"flex"}}>
```
Comment (sraghudatta94) [inline src/components/Dashboard/Admin/Dashboard.js:49]: try to minimize inline css

File: src/components/Dashboard/Admin/Dashboard.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+                                <img src={review.profile} alt="profile" />
+                                <p>{review.name}<br/>{ review.comment}</p>
```
Comment (sraghudatta94) [inline src/components/Dashboard/Admin/Dashboard.js:51]: code destructuring

File: src/components/Dashboard/Common/Header/DashHeader.css [hunk 4]
```diff
@@ -0,0 +1,17 @@
+.header{
+    width: 80.5vw;
```
Comment (sraghudatta94) [inline src/components/Dashboard/Common/Header/DashHeader.css:2]: make it consistent either % or px/em/vw

File: src/components/Dashboard/Common/Quiz/Quiz.js [hunk 5]
```diff
@@ -0,0 +1,57 @@
+import React, { useState} from 'react'
+import DashHeader from '../Header/DashHeader'
+import { QuestionOutlined }  from "@ant-design/icons"
+import { Input,Card,Button } from "antd"
+import "./Quiz.css"
+import { quizList,resourceFilter } from '../../../../constant/data'
```
Comment (sraghudatta94) [inline src/components/Dashboard/Common/Quiz/Quiz.js:6]: absolute imports pls

File: src/components/Dashboard/Common/Quiz/Quiz.js [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+    }
+    
+    const onSearch = (value) => {
+        let filtered = quizList.filter(res => res.title.includes(value));
+        setQuiz(filtered)
+    };
+
+    return (
+        <div className="quiz-container">
+            <DashHeader icon={<QuestionOutlined />} title="Quiz" rightComponent={<>
```
Comment (sraghudatta94) [inline src/components/Dashboard/Common/Quiz/Quiz.js:33]: make an icons.js const and put it in assets folder. use that everywhere

File: src/components/Dashboard/Common/Resource/Resource.css [hunk 7]
```diff
@@ -0,0 +1,31 @@
+/* .resource{
+    width:82.5vw;
```
Comment (sraghudatta94) [inline src/components/Dashboard/Common/Resource/Resource.css:2]: remove dead code

File: src/components/Dashboard/Interviewer/Profile/InterviewerProfile.css [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+}
+
+.right-profile{
+    text-align: left;
+    padding: 7vh 0;
+    width: 50vw;
+}
+
+.update{
+        background-color: royalblue;
```
Comment (sraghudatta94) [inline src/components/Dashboard/Interviewer/Profile/InterviewerProfile.css:40]: always use #

File: src/components/Dashboard/Student/List/StudentList.js [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+                    <>
+                        <Filter filterOptions={studentFilter} filterFunction={handleFilter} placeholder="Filter Students" />
+                        <section className="search">
+                            <Search placeholder="Search Students"  onSearch={onSearch} style={{ width: 200 }} />
+                        </section>
+                   
+                    </>} />
+                    <div className="students">
+
+                    {studentList.length > 0 ?
```
Comment (sraghudatta94) [inline src/components/Dashboard/Student/List/StudentList.js:47]: recheck if this is required pls

Comment (sraghudatta94) [reply src/components/Dashboard/Student/List/StudentList.js:47]: x?y:z

File: src/components/Dashboard/Student/List/StudentList.js [hunk 10]
```diff
@@ -0,0 +1,97 @@
...
+                        <> {
+                            studentList.map(student => (
+                                <section key={student.id} className="listedInterviewer">
```
Comment (sraghudatta94) [inline src/components/Dashboard/Student/List/StudentList.js:50]: pls make this as a section component

File: src/components/Dashboard/Student/Profile/StudentProfile.css [hunk 11]
```diff
@@ -0,0 +1,23 @@
+.student-profile{
+    width: 83.5;
+    position: absolute;
+    right: 0;
+    box-sizing: border-box;
+
+}
+
+.right-profile>.about{
+    height: auto !important;
```
Comment (sraghudatta94) [inline src/components/Dashboard/Student/Profile/StudentProfile.css:10]: i see important in many places in many css files. please recheck once (low priority)

File: src/components/Dashboard/Student/Profile/StudentProfile.js [hunk 12]
```diff
@@ -0,0 +1,94 @@
...
+import { InterviewerForm } from "../../../Primary Components/Forrm/Interviewer/InterviewerForm"
+import { Progress } from 'antd';
+import "./StudentProfile.css"
+
+export const StudentProfile = () => {
+
+    const student = students[0];
+    const [isModalVisible, setIsModalVisible] = useState(false);
+
+    const studentForm = JSON.parse(JSON.stringify(student));
```
Comment (sraghudatta94) [inline src/components/Dashboard/Student/Profile/StudentProfile.js:18]: pls check if there is a better way

File: src/components/Join Us/JoinUs.js [hunk 13]
```diff
@@ -0,0 +1,25 @@
...
+
+export default function JoinUs ({isSignUp}){
+    return (
+        <>
+             <MPHeader/>
+            <div className="Join">
+            
+                <section className="signBox">
+                    <img src={logoUrl} alt="profile" />
+                    <JoinUsButton className="sign-btn" signin={isSignUp} type={"facebook"}  />
```
Comment (sraghudatta94) [inline src/components/Join Us/JoinUs.js:16]: we decided to hold fb login

File: src/components/Join Us/SignIn/SignIn.js [hunk 14]
```diff
@@ -0,0 +1,55 @@
...
+
+                    <img src={logoUrl} alt="profile" />
+
+                    <section className="dummy">
+                        <Button className="dummyToggle" onClick={normalSign}>Sign In</Button>
+                        <Button className="dummyToggle" onClick={dummySign}>Dummy Sign In</Button>
+                    </section>
+
+                    {dummy ?
+                    <section className="dummyButtons">
```
Comment (sraghudatta94) [inline src/components/Join Us/SignIn/SignIn.js:36]: tab view not button view pls

File: src/components/Packages/Packages.js [hunk 15]
```diff
@@ -0,0 +1,25 @@
...
+    return (<>
+            <MPHeader/>
+            <section className="package-detail">
+                {packages.map(pack => (
+                    <section key={ pack.id} className="pack">
+                        <h2>{pack.title}</h2>
+                        <p>{pack.description}</p>
+                        <ul>
+                            {pack.details.map(detail => (
+                                <li key={detail}>{detail}</li>
```
Comment (sraghudatta94) [inline src/components/Packages/Packages.js:16]: li key should be `key`

File: src/components/Primary Components/Button/MP_Button.js [hunk 16]
```diff
@@ -0,0 +1,7 @@
+import './MP_Button.css'
+import { Button } from 'antd';
+
+export default function MPButton(props){
+    return <Button type={props.buttonType} onClick={props.onClick} disabled={!props.isDisabled} size = {props.size} id="mainButton" className={props.className} style={{width:props.width}}>{props.buttonName}</Button>
```
Comment (sraghudatta94) [inline src/components/Primary Components/Button/MP_Button.js:5]: always destructure functions

File: src/components/Primary Components/Footer/Footer.css [hunk 17]
```diff
@@ -0,0 +1,41 @@
...
+}
+
+footer>section>p{
+    color: white;
+    margin: 5vh;
+    margin-bottom: 0 !important ;
+}
+
+footer>section>h2{
+    color:white;
```
Comment (sraghudatta94) [inline src/components/Primary Components/Footer/Footer.css:28]: Hex color code

File: src/components/Primary Components/Forrm/Interviewer/InterviewerForm.js [hunk 18]
```diff
@@ -0,0 +1,44 @@
...
+    const firstCol = keys.splice(0, middleIndex);   
+    const secondCol = keys.splice(-middleIndex);
+  
+    return (
+        <div>
+            <Form  name="control-hooks form" onFinish={onFinish} autoComplete="on">
+                <Row gutter={24}>
+                    <Col span={12} style={{textAlign: 'left'}}>
+                        {firstCol.map((key,index) => (
+                            <Form.Item key={ index} className="form-input"  name={key}  label={key.toUpperCase()}  rules={[{ required: false, message: 'Please input your username!' }]}>
```
Comment (sraghudatta94) [inline src/components/Primary Components/Forrm/Interviewer/InterviewerForm.js:21]: Please make rules right. if possible make it to a json const and reuse here

File: src/components/Primary Components/Join Us/Join Us Button.css [hunk 19]
```diff
@@ -0,0 +1,33 @@
+.JoinUsButton{
+    /* display: flex;
+    align-items: center;
+    justify-content: center;
```
Comment (sraghudatta94) [inline src/components/Primary Components/Join Us/Join Us Button.css:4]: no dead code pls

File: src/components/Primary Components/Join Us/Join Us Button.js [hunk 20]
```diff
@@ -0,0 +1,25 @@
...
+
+const details = {
+    google : <GoogleOutlined className='icon' style={{ fontSize: '40px', color: '#EA4335' }} />,
+    facebook : <FacebookFilled className='icon' style={{ fontSize: '40px', color: '#3B5998' }} />
+}
+
+export default function JoinUsButton ({signin,type}) {
+
+    function getUserDetails(){
+      alert("Login Clicked")
```
Comment (sraghudatta94) [inline src/components/Primary Components/Join Us/Join Us Button.js:12]: no default alerts

**Review by sraghudatta94 — CHANGES_REQUESTED** [review]

**Comment by saifmohammed888** [comment]
Week1 Changes
Week2 Changes

**Pull request merged**